docs/graph_engine/overview.rst

```diff
@@ -0,0 +1,194 @@
+  of shape `(len(nodes), sum(map(lambda f: f[1], features)))` where features values are placed in the same order
+  as in the features list. It is also used to fetch labels for supervised training.
+
+  Example: how to extract multiple features with different dimensions in a single graph call. Lets extract
+  feature with *id 0* and dimension **64**, multiply by some matrix with dim **64x32** (so the result is
+  **1x32** vector) and then do some aggregation via a dot product on a feature vector with *id 1* and dimension
+  **32**, so the result is a single number per node.
+
+  .. code-block:: python
+
+      >>> nodes = graph.sample_nodes(10, 0)
+      >>> weights = np.random.rand(64,32)
+      >>> features = graph.node_features(nodes, [(0, 64), (1, 32)], FeatureType.float)
+      >>> embed = torch.mm(features[:,0:64], weights)
+      >>> result = torch.mm(embed, features[:, 64:])
+
+
+- :code:`edge_features(edges: np.array, features: List[(id, dim)], feature_type: FeatureType) -> np.array`
+
+  Similar function to node_features, but for edges.
+
+- :code:`node_count(types: Union[int, np.array]) -> int`
+
+  Return the number of nodes for the specific node types.
+
+- :code:`edge_count(types: Union[int, np.array]) -> int`
+
+  Return the number of edge for the specific edge types.
+
+- :code:`random_walk(self, nodes: np.array, edge_types: np.array, walk_len: int, p: float, q: float, default_node: int = -1, seed: Optional[int] = None) -> np.array`
+
+  Execute a random walk starting with nodes given. A random walk is equivalent to repeatedly calling sample_neighbors walk_len times, each time setting nodes to be the newly sampled neighbors.
+
+  For the p return parameter, 1/p is the probability to return a parent node.
+  For the q in-out parameter, 1/q is the probability to select a neighbor not connected to a parent.
+
+  Return is ndarray shape [nodes.size, walk_len+1] as the original nodes start the sampled walk.
+
+Example
+-------
+
+Lets explore a contrived example, we generated all the neighbors offline from a database and put them in a
+following record:
+
+.. code-block:: python
+
+    >>> class Record:
+    ...    """Represent a row of features from a database"""
+    ...    def __init__(self, f:np.array, fn:List[np.array], sn:[List[List[np.array]]]):
+    ...        self.nodeFeatures = f
+    ...        self.fstHopNeighborFeatures = fn
+    ...        self.sndHopNeighborFeatures = sn
+
+
+One interesting thing about current GNN algorithms is the `node_id` is not saved anywhere in a model and we can
+generate fake node_ids as long as they are consistent with the original graph(fake ids map to exactly one real
+node). Here is a possible implementation of the API where for neighbor sampling we are going to generate fake
+node_ids that can be used to uniquely identify features in a record list:
+
+.. code-block:: python
+
+    >>> class RecordGraph:
```

Comment:
If someone copies this, will it run? We should add doctest block below.

Reply:
It can be removed actually, because there is a dedicated doctest example.